### PR TITLE
GCP Pub/Sub - Topic creation and Project configuration enhancements

### PR DESCRIPTION
**Pull request by silviu-negoita**
- Creating Subscription / Topic is configurable
- Support project-id conf at connector level - backing to global one

### CODE REVIEW

File: attic/smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java [hunk 1]
```diff
@@ -92,17 +97,28 @@ public void destroy(@Observes @Destroyed(ApplicationScoped.class) final Object c
...
                 .ignore();
     }
 
+    private String getProjectId(Config config) {
+        return config.getOptionalValue("project-id", String.class)
+                .orElse(projectId);
+    }
+
+    boolean isUseAdminClient(Config config) {
+        return config.getOptionalValue("use-admin-client", Boolean.class).orElse(true);
```
Comment (cescoffier) [inline attic/smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java:119]: At some point, we really need to switch to @ConnectorAttribute and avoid that kind of parsing.

File: attic/smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java [hunk 2]
```diff
@@ -92,17 +97,28 @@ public void destroy(@Observes @Destroyed(ApplicationScoped.class) final Object c
...
     public SubscriberBuilder<? extends Message<?>, Void> getSubscriberBuilder(final Config config) {
-        final PubSubConfig pubSubConfig = new PubSubConfig(projectId, getTopic(config), getCredentialPath(config),
+        final PubSubConfig pubSubConfig = new PubSubConfig(getProjectId(config), getTopic(config), getCredentialPath(config),
                 mockPubSubTopics, host.orElse(null), port.orElse(null));
 
         return ReactiveStreams.<Message<?>> builder()
                 .flatMapCompletionStage(message -> CompletableFuture.supplyAsync(() -> {
-                    createTopic(pubSubConfig);
+                    if (isUseAdminClient(config)) {
+                        createTopic(pubSubConfig);
```
Comment (cescoffier) [inline attic/smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java:106]: I would add an INFO message indicating that the topic is going to be created if needed.

**Review by cescoffier — CHANGES_REQUESTED** [review]
The code of the PR looks good, I've made a minor comment.

Can you move back the connector outside of the attic? 

1. Move the directory to the project root
2. Add the module to the root `pom.xml` file

File: attic/smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java (hunk 1 above)
Comment (silviu-negoita) [reply attic/smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java:119]: I can do that in future. You can create a task / issue and assign to me when u want.

File: attic/smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java (hunk 2 above)
Comment (silviu-negoita) [reply attic/smallrye-reactive-messaging-gcp-pubsub/src/main/java/io/smallrye/reactive/messaging/gcp/pubsub/PubSubConnector.java:108]: check `log.adminClientEnabled();`

**Review by cescoffier — APPROVED** [review]
Thanks!

**Comment by silviu-negoita** [comment]
@cescoffier i see the test are not compiling anymore. some ideas?
```
2021-11-02T13:09:10.0739869Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 46.548 s <<< FAILURE! - in io.smallrye.reactive.messaging.gcp.pubsub.PubSubManagerTest
2021-11-02T13:09:10.0774756Z [ERROR] testResourceCleanup  Time elapsed: 0.96 s  <<< ERROR!
2021-11-02T13:09:10.0775956Z org.jboss.weld.exceptions.DeploymentException: 
2021-11-02T13:09:10.0777832Z WELD-001408: Unsatisfied dependencies for type ConnectorFactories with qualifiers @Default
2021-11-02T13:09:10.0835365Z   at injection point [BackedAnnotatedParameter] Parameter 1 of [BackedAnnotatedConstructor] @Inject public io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory(ConnectorFactories, Instance<Config>, @Any Instance<ChannelRegistry>)
2021-11-02T13:09:10.0851967Z   at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.<init>(ConfiguredChannelFactory.java:0)
```

**Comment by cescoffier** [comment]
You need to add the `ConnectorFactories` class to the set of deployed classes (when we initialize Weld)

**Comment by silviu-negoita** [comment]
> ConnectorFactories

Fixed. thanks!

**Comment by silviu-negoita** [comment]
@cescoffier can u please approve again?

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
The PR looks good and is ready to be merged. I will orchestrate the merge with @ozangunalp as we have a few releases to do in the next few days. It will likely be in one of them (but maybe not the first one; again no panic, it will be there)

**Comment by cescoffier** [comment]
As promised: merged! Thanks!

**Pull request merged**